numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -45,14 +45,20 @@ static int
 raise_binary_type_reso_error(PyUFuncObject *ufunc, PyArrayObject **operands) {
     PyObject *errmsg;
     const char *ufunc_name = ufunc_get_name_cstr(ufunc);
-    errmsg = PyUString_FromFormat("ufunc %s cannot use operands "
-                        "with types ", ufunc_name);
-    PyUString_ConcatAndDel(&errmsg,
-            PyObject_Repr((PyObject *)PyArray_DESCR(operands[0])));
-    PyUString_ConcatAndDel(&errmsg,
-            PyUString_FromString(" and "));
-    PyUString_ConcatAndDel(&errmsg,
-            PyObject_Repr((PyObject *)PyArray_DESCR(operands[1])));
+    errmsg = PyUString_FromFormat(
+        "ufunc %s cannot use operands with types {!r} and {!r}", ufunc_name);
+    if (errmsg == NULL) {
+        return -1;
+    }
+    Py_SETREF(
+        errmsg, PyObject_CallMethodObjArgs(errmsg, "format",
+            (PyObject *)PyArray_DESCR(operands[0]),
+            (PyObject *)PyArray_DESCR(operands[1])
+        )
```

Comment:
[PyObject_CallMethodObjArgs](https://docs.python.org/2.7/c-api/object.html#c.PyObject_CallMethodObjArgs) cannot take a `const char *`  name, it must be a `PyObject*`. Were you thinking of `PyObject_CallMethod` ? Perhaps this can be a helper function

Reply:
Ugh, of course this was too easy.